core/src/main/scala/com/ubirch/user/core/manager/UserManager.scala

```diff
@@ -187,6 +188,29 @@ object UserManager extends StrictLogging
       }
   }
 
+  def findByUsedIds(userIds: Seq[UUID])
+                       (implicit mongo: MongoUtil): Future[Seq[User]] = {
+
+    val selector = document("id" -> document("$in" -> userIds.map(_.toString)))
+
+    mongo
+      .collection(collectionName)
+      .flatMap { collection =>
+        collection
+          .find[BSONDocument, User](selector)
+          .cursor[User]()
+          .collect[Seq](
+            -1,
+            Cursor.FailOnError[Seq[User]]()
+          )
+      }
+      .recover {
+        case ex =>
+          logger.error(s"error when retrieving users with the following userIds $userIds: ${ex.getMessage}")
```

Comment:
I prefer nowadays to propagate the error to the http response. Otherwise the client can be mistaking that none of the users exist. This is a weird style from the old times, everywhere still alive.. ;)

Reply:
makes sense